### PR TITLE
Updated library interfaces.

### PR DESCRIPTION
**Pull request by chhsia0**

### CODE REVIEW

**Comment by jbarrick-mesosphere** [comment]
FYI you can run `make build` to regenerate the docs

Do we maybe want to add a linter for pep8?

**Review by jbarrick-mesosphere — APPROVED** [review]

**Comment by masonse** [comment]
> FYI you can run `make build` to regenerate the docs
> 
> Do we maybe want to add a linter for pep8?

@jbarrick-mesosphere are you thinking of https://pypi.org/project/pep8/

**Comment by jbarrick-mesosphere** [comment]
Yeah.

File: starlark/experimental/buildkit.star [hunk 1]
```diff
@@ -10,105 +10,123 @@ Provides methods for interacting with a Buildkit instance.
...
+    --frontend=dockerfile.v0 \
+    --local context=/ \
+    --local dockerfile=/tmp \
+    --output type=local,dest=/
+    --opt filename=Dockerfile.buildkit
+        """.format(dockerfile)],
+        **kwargs
+    )
+
+def buildkit(task_name, git_name, image_repo, tag="$(context.build.name)", context=".", dockerfile="Dockerfile", build_args={}, build_env={}, **kwargs):
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/buildkit.star]: We should update the `kaniko` method signature too

**Comment by jbarrick-mesosphere** [comment]
I noticed we didn't change the `pullRequest` method to snake case

File: starlark/experimental/buildkit.star [hunk 2]
```diff
@@ -10,105 +10,123 @@ Provides methods for interacting with a Buildkit instance.
...
-    build_args = []
-
-    for k, v in (buildArgs or {}).items():
-      build_args += ["--opt", "build-arg:{}={}".format(k, v)]
-
-    for k, v in (buildEnv or {}).items():
-      build_args += ["--opt", "build-env:{}={}".format(k, v)]
-
-    task(name, inputs = [git] + (inputs or []), outputs = [name], steps=(steps or []) + [
+    image_name = image_resource("image-" + sanitize(image_repo)[-57:], url = image_repo + ":" + tag)
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/buildkit.star]: Should sanitize trim the string too?

File: starlark/experimental/buildkit.star [hunk 3]
```diff
@@ -10,105 +10,120 @@ Provides methods for interacting with a Buildkit instance.
...
+{}
+EOF
+buildctl build \
+    --debug \
+    --addr=tcp://buildkitd.buildkit:1234 \
+    --progress=plain \
+    --frontend=dockerfile.v0 \
+    --local context=/ \
+    --local dockerfile=/tmp \
+    --output type=local,dest=/
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/buildkit.star]:
```suggestion
    --output type=local,dest=/ \
```

File: starlark/experimental/buildkit.star [hunk 4]
```diff
@@ -10,105 +10,120 @@ Provides methods for interacting with a Buildkit instance.
 To import, add the following to your Dispatchfile:
 
 ```
-load("github.com/mesosphere/dispatch-catalog/starlark/experimental/buildkit@0.0.4", "buildkit")
+load("github.com/mesosphere/dispatch-catalog/starlark/experimental/buildkit@chhsiao/dsl-name", "buildkit")
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/buildkit.star]:
```suggestion
load("/starlark/experimental/buildkit", "buildkit")
```

File: starlark/experimental/buildkit.star [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # vi:syntax=python
 
-load("github.com/mesosphere/dispatch-catalog/starlark/stable/pipeline@master", "clean", "imageResource", "volume")
+load("github.com/mesosphere/dispatch-catalog/starlark/stable/pipeline@chhsiao/dsl-name", "sanitize", "image_resource", "git_checkout_dir")
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/buildkit.star]:
```suggestion
load("/starlark/stable/pipeline", "sanitize", "image_resource", "git_checkout_dir")
```

File: starlark/experimental/go.star [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # vi:syntax=python
 
-load("github.com/mesosphere/dispatch-catalog/starlark/stable/pipeline@master", "imageResource", "storageResource", "resourceVar")
-load("github.com/mesosphere/dispatch-catalog/starlark/experimental/buildkit@master", "buildkitContainer")
+load("github.com/mesosphere/dispatch-catalog/starlark/stable/pipeline@chhsiao/dsl-name", "git_checkout_dir", "image_resource", "sanitize", "storageResource", "resourceVar")
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/go.star]:
```suggestion
load("/starlark/stable/pipeline", "git_checkout_dir", "image_resource", "sanitize", "storageResource", "resourceVar")
```

File: starlark/experimental/go.star [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
+load("github.com/mesosphere/dispatch-catalog/starlark/experimental/buildkit@chhsiao/dsl-name", "buildkit_container")
```
Comment (jbarrick-mesosphere) [inline starlark/experimental/go.star]:
```suggestion
load("/starlark/experimental/buildkit", "buildkit_container")
```

File: starlark/experimental/buildkit.star (hunk 4 above)
Comment (chhsia0) [reply starlark/experimental/buildkit.star]: This is cool!

**Review by takirala — APPROVED** [review]

**Pull request merged**